components/automate-ui/src/app/components/sidebar/sidebar.component.ts

```diff
@@ -15,12 +15,36 @@ export class SidebarComponent {
   constructor(
     @Inject(LayoutFacadeService) public layoutFacade: LayoutFacadeService
   ) {
-    this.menuGroups$ = layoutFacade.menuGroups$;
+    this.menuGroups$ = layoutFacade.sidebar$;
+    this.updateMenuGroupVisibility();
   }
 
-  public isVisible(item: any) {
-    return item.subscribe
-      ? item
-      : new BehaviorSubject(item);
+  public updateMenuGroupVisibility(): void {
+    this.menuGroups$.subscribe((menuGroups: MenuItemGroup[]) => {
+      if (menuGroups) {
+        menuGroups.forEach(menuItemGroup => this.hasVisibleMenuItems(menuItemGroup));
+      }
+    });
+  }
+
+  public hasVisibleMenuItems(menuItemGroup: any): void {
+    menuItemGroup.hasAuthorizedMenuItems =
+      this.hasAuthorizedMenuItems(menuItemGroup) ||
+      this.hasNoAuthorizedMenuItems(menuItemGroup);
+  }
```

Comment:
The logic is perfectly sound here, but the naming confused me for a bit. After several passes I came up with this to improve clarity; what do you think?

- hasVisibleMenuItems => setGroupVisibility
- hasAuthorizedMenuItems => hasVisibleMenuItems
- hasNoAuthorizedMenuItems => hasMenuItemsNotRequiringAuthorization

resulting in:
```
  public setGroupVisibility(menuItemGroup: MenuItemGroup): void {
    menuItemGroup.hasVisibleMenuItems =
      this.hasAuthorizedMenuItems(menuItemGroup) ||
      this.hasMenuItemsNotRequiringAuthorization(menuItemGroup);
  }
```
(Also added type instead of `any`, but then you will need to update the definition of MenuItemGroup to include hasVisibleMenuItems.)

Happy to chat about these suggested changes as this is of key importance, I believe.

Reply:
@msorens Updated change request 😊